galsim/scene.py

```diff
@@ -568,6 +581,87 @@ def makeGalaxy(self, index=None, gal_type=None, chromatic=False, noise_pad_size=
         else:
             return gal_list[0]
 
+    def selectRandomIndex(self, n_random=1, rng=None, _n_rng_calls=False):
```

Comment:
Good catch.  Is what I did just now in https://github.com/GalSim-developers/GalSim/pull/854/commits/958b8e98fabf46e8d4c6d4a0c0651e7bb64c7fd5 what you had in mind?  Or did you also have in mind to add a method of BaseDeviate that calls this utility?